manifests/agent.pp

```diff
@@ -314,16 +314,9 @@
   }
 
   # Ensure that the correct config file is used.
-  
-  if $::osfamily == 'debian' {
-    if versioncmp($::operatingsystemmajrelease, '8') == -1 {
-      file { '/etc/init.d/zabbix-agent':
-        ensure  => file,
-        mode    => '0755',
-        require => Package[$zabbix_package_agent],
-        content => template('zabbix/zabbix-agent-debian.init.erb'),
-      }
-    } else {
+
+  if $::osfamily in ['Debian', 'RedHat'] {
```

Comment:
I think we can remove that if? There are multiple other distributions that also ship systemd. the next if with any2bool() should be enough